packages/common/src/services/submissionService.ts

```diff
@@ -211,6 +255,72 @@ const service = (dependencies: Dependencies) => {
 
 			return parseActiveSubmissionResponse(submission);
 		},
+
+		commitSubmission: async (categoryId: number, submissionId: number): Promise<CommitSubmissionResult> => {
+			const { getSubmissionById } = submissionRepository(dependencies);
+			const { getSubmittedDataByCategoryId } = submittedRepository(dependencies);
+			const { getActiveDictionaryByCategory } = categoryRepository(dependencies);
+
+			const submission = await getSubmissionById(submissionId);
+			if (_.isEmpty(submission) || !submission.dictionaryId)
+				throw new BadRequest(`Submission '${submissionId}' not found`);
+
+			if (submission.dictionaryCategoryId !== categoryId)
+				throw new BadRequest(`Category ID provided does not match the category for the Submission`);
+
+			if (submission.state !== SUBMISSION_STATE.VALID)
+				throw new StateConflict('Submission does not have state VALID and cannot be committed');
+
+			if (!categoryId) throw new BadRequest(`Active Submission does not belong to any Category`);
+
+			const currentDictionary = await getActiveDictionaryByCategory(categoryId);
+			if (_.isEmpty(currentDictionary)) throw new BadRequest(`Dictionary in category '${categoryId}' not found`);
+
+			const entitiesToProcess: string[] = [];
+
+			const submittedDataArray = await getSubmittedDataByCategoryId(categoryId);
+
+			const submissionsToValidate = Object.entries(submission.data).flatMap(([entityName, submissionEntity]) => {
+				entitiesToProcess.push(entityName);
+				return submissionEntity.records.map((record) => {
+					const newSubmittedData: NewSubmittedData = {
+						data: record,
+						dictionaryCategoryId: categoryId,
+						entityName,
+						organization: submission.organization,
+						originalSchemaId: submission.dictionaryId,
+						lastValidSchemaId: submission.dictionaryId,
+						createdBy: '', // TODO: get User from auth
+					};
+					return newSubmittedData;
+				});
+			}, {});
+
+			if (_.isArray(submittedDataArray) && submittedDataArray.length > 0) {
```

Comment:
other than perhaps being a tiny bit faster, how is using `_.isArray` better than `Array.isArray`?

Reply:
There is no significant reason to use _.isArray. In fact I change it to use the built-in isArray.